### PR TITLE
Unify filter feature and improve space management for EditorFileDialog

### PR DESCRIPTION
**Pull request by vPumpking**
Move buttons related to the file view on the container above so that they take less horizontal space and make search bar similar to the one in FileDialog

This is part 2 of proposal https://github.com/godotengine/godot-proposals/issues/9099
requested by @Calinou after https://github.com/godotengine/godot/pull/88673#pullrequestreview-2340996679

see https://github.com/godotengine/godot/pull/97344 and https://github.com/godotengine/godot/pull/88673
replaces PR #97941 which was based on a forked branch instead of upstream

### CODE REVIEW

**Comment by vPumpking** [comment]
feature is ready

**Comment by Calinou** [comment]
There are some merge conflicts, could you rebase on top of `master` so I can review this?

**Comment by vPumpking** [comment]
sure, tomorrow that'll be done

**Review by Calinou — APPROVED** [review]
Tested locally, it works as expected. Code looks good to me.

While testing this PR, I found an additional usability issue: https://github.com/godotengine/godot/issues/98838

**Review by AThousandShips — APPROVED** [review]

File: editor/gui/editor_file_dialog.cpp [hunk 1]
```diff
@@ -1146,6 +1148,15 @@ void EditorFileDialog::_filter_selected(int) {
 	update_file_list();
 }
 
+void EditorFileDialog::_search_filter_selected() {
+	Vector<int> items = item_list->get_selected_items();
+	if (!items.is_empty()) {
+		int index = items[0];
+		file->set_text(item_list->get_item_text(index));
+		file->emit_signal("text_submitted", file->get_text());
```
Comment (AThousandShips) [inline editor/gui/editor_file_dialog.cpp:1156]:
```suggestion
		file->emit_signal(SceneStringName(text_submitted), file->get_text());
```
New constant

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**